### DISCUSSION TITLE
Invalid <Link> with <a> child error is driving me nuts

### DISCUSSION THREAD

**Discussion by Isaac-Tait** [vercel/next.js#42994] [Help] [answered]
Since upgrading to Next 13 I keep getting the error message:

```
Error: Invalid <Link> with <a> child. Please remove <a> or use <Link legacyBehavior>.
Learn more: https://nextjs.org/docs/messages/invalid-new-link-with-extra-anchor
```

So, I went through my site with a fine tooth comb and cleaned up the `<Link>s` with an `<a>` tag as dictated by the docs [here](https://nextjs.org/docs/messages/invalid-new-link-with-extra-anchor). Then I ran the code mod just to be sure I had not missed anything (I had - in my header component [thanks code mod]). 

Now my site loads in localhost (it used to throw the error right away) however when I attempt to navigate to any sub-pages I keep getting the `Invalid <Link> with <a> child` error. The problem with the error is that it does not specify where I have an incorrect `<Link>` tag. When I run the code mod it returns:

```
All done.
Results:
0 errors
27 unmodified
0 skipped
5 ok
```

The site builds fine in Vercel, so there are no build errors - just in local host. My repo can be found [here](https://github.com/Isaac-Tait/ilovetofish)

Has anyone else run into this issue? If so how did you resolve it? I did notice that the code mod was putting `legacyBehavior` as a prop on `<Link>` tags without an `<a>` tag (as can be seen [here](https://github.com/Isaac-Tait/ilovetofish/commit/987a6b61c21e6053ec3eba72699c3341c8576134)) - it is my understanding that you do not need the `legacyBehavior` prop unless you are using an `<a>` tag. Is that correct?

**Comment by Isaac-Tait** [comment, marked as answer]
I figured out what was going on [nextjs-breadcrumbs](https://www.npmjs.com/package/nextjs-breadcrumbs) is not supporting Next 13 yet... 

https://github.com/NiklasMencke/nextjs-breadcrumbs/issues/47

**Reply by icyJoseph** [reply]
Oh boy, you know I checked out your repo and found nothing so I was like... hope for the best. Good stuff though! Can't forget 3rd party libraries! You could deactivate the behaviour through next.config.js though.

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  experimental: {
    newNextLinkBehavior: false,
  },
};

module.exports = nextConfig;

```

So you can take the 13 update anyway. In this phase, the new link behavior is opt-out, so it should be ok until all other parts are up to date.

**Reply by digitalappsbd** [reply]
wow...great...thanks!

**Reply by muhammadmairajj** [reply]
Problem solved,
Thank you

**Reply by yubliwarokkaid** [reply]
![Screenshot_4](https://user-images.githubusercontent.com/72744397/236678538-ccd5dae7-0c30-41a2-aa94-1830d4d05178.png)

**Reply by Kofichamp** [reply]
Thanks pals

**Reply by VictorYarosh** [reply]
Thanks )))

**Comment by Isaac-Tait** [comment]
@icyJoseph 

> hope for the best 🤣

Thanks though for taking a look at it, I appreciate it - and thank you for the `next.config.js` tip too, I will implement that. Cheers!

**Comment by jserrao** [comment]
Yeah similar situation - the error docs are useless when you're upgrading a large codebase. You just have to find/replace and comb through everything - and you'll probably miss a couple so good luck tracking them down.

**Comment by a-k-holod** [comment]
_Error: Invalid <Link> with <a> child. Please remove <a> or use <Link legacyBehavior>.
Learn more: https://nextjs.org/docs/messages/invalid-new-link-with-extra-anchor_

- When following this link, there is solution:
- `npx @next/codemod new-link .`

No need to change tags by hand.
